hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/placement/algorithms/SCMContainerPlacementRackAware.java

```diff
@@ -561,7 +561,7 @@ private List<DatanodeDetails> chooseNodes(List<DatanodeDetails> excludedNodes,
       DatanodeDetails favoredNode = favoredNodeNum > favorIndex ?
           favoredNodes.get(favorIndex) : null;
       DatanodeDetails chosenNode;
-      if (favoredNode != null && networkTopology.isSameParent(
+      if (favoredNode != null && !networkTopology.isSameParent(
```

Comment:
I guess usedNodes also needs to be considered here & not just excludedNodes & chosenNodes

Reply:
@swamirishi Thanks for the review.
UsedNodes are already added into [excludedNodes](https://github.com/apache/ozone/blob/b3c84845926fb79e4b3b4ae20ded2be1c97ada06/hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/placement/algorithms/SCMContainerPlacementRackAware.java#L227) in the caller. And nodes here we are choosing in different rack than excluded nodes. So it's considering usedNodes as well here.